docs/canvas/canvas-function-reference.asciidoc

```diff
@@ -1,17 +1,2133 @@
+expression in your argument, Canvas treats that argument as a measure. Otherwise, it 
+is a dimension. Dimensions are combined to create unique keys. Measures are 
+then deduplicated by those keys using the specified TinyMath function.
+
+*Accepts:* `datatable`
+
+[cols="3*^<"]
+|===
+|Argument |Type |Description
+
+|`x`
+|`string`
+|The values along the X-axis
+
+|`y`
+|`string`
+|The values along the Y-axis
+
+|`color`
+|`string`
+|An expression to use in determining the mark's color
+
+|`size`
+|`string`
+|For elements that support it, the size of the marks
+
+|`text`
+|`string`
+|For use in charts that support it, the text to show in the mark
+|===
+
+*Returns:* `pointseries`
+
+
+[float]
+[[progress_fn]]
+=== `progress`
+
+Configure a progress element
+
+*Accepts:* `number`
+
+[cols="3*^<"]
+|===
+|Argument |Type |Description
+
+|`barColor`
+|`string`
+|Color of the background bar
+
+Default: `"#f0f0f0"`
+
+|`barWeight`
+|`number`
+|Thickness of the background bar
+
+Default: `20`
+
+|`font`
+|`style`
+|Font settings for the label. Technically you can stick other styles in here too
+
+Default: `{<<font_fn>> size=24 family="'Open Sans', Helvetica, Arial, sans-serif" color="#000000" align=center}`
+
+|`label`
+|`boolean`, `string`
+|Set `true`/`false` to show/hide label or provide a string to display as the label
+
+Default: `true`
+   
+|`max`
+|`number`
+|Maximum value of the progress element
+
+Default: `1`
+
+|`shape`
+|`string`
+|Select {list}, or {end}`,
+      values: {
+        list: Object.values(Shape)
+          .slice(0, -1)
+          .join(', '),
+        end: Object.values(Shape).slice(-1)[0],
+      },
+
+Default: `"gauge"`
+
+|`valueColor`
+|`string`
+|Color of the progress bar
+
+Default: `"#1785b0"`
+
+|`valueWeight`
+|`number`
+|Thickness of the progress bar
+
+Default: `20`
+|===
+
+*Returns:* `render`
+
+
+[float]
+[[r_fns]]
+== R
+[[render_fn]]
+=== `render`
+
+Renders an input as a specific element and sets element level options,
+such as styling.
+
+*Accepts:* `render`
+
+[cols="3*^<"]
+|===
+|Argument |Type |Description
+
+|`as`
+|`string`
+|The element type to use in rendering. You might want to use a
+specialized function instead, such as `<<plot_fn>>` or `debug`.
+
+|`css`
+|`string`
+|Any block of custom CSS to be scoped to this element 
+
+Default: `".canvasRenderEl{\n\n}"`
+
+|`containerStyle`
+|`containerStyle`
+|Style for the container, including background, border, and opacity
+
+Default: `{<<containerStyle_fn>>}`
+|===
+
+*Returns:* `render`
+
+
+[float]
+[[repeatImage_fn]]
+=== `repeatImage`
+
+Configures a repeating image element.
+
+*Accepts:* `number`
+
+[cols="3*^<"]
+|===
+|Argument |Type |Description
+
+|`image`
+|`string`
+|An image asset can be provided as a base64 dataurl, or passed in as a subexpression
+ (see example below)
+
+|`size`
+|`number`
+|The maximum height or width of the image, in pixels. Eg, if you images is taller than it is wide, this will limit its height 
```

Comment:
How about:

`When the image is taller than it is wide, this function limits the height.`